includes/import.form.inc

```diff
@@ -1240,3 +1286,185 @@ function islandora_multi_importer_startprocessing($parameters) {
   // islandora-multi-importer-zeroMQboot > /dev/null &");
   // islandora_multi_importer_client_task();
 }
+
+/**
+ * Implements hook_form_alter.
+ *
+ * This function implements hook_form_alter with intent to set defaults
+ * for islandora_multi_importer_form fields.
+ *
+ * @param $form
+ * @param $form_state
+ * @param $form_id
+ */
+function islandora_multi_importer_form_alter(&$form, &$form_state, $form_id) {
+
+  // The big switch...
+  switch($form_id) {
+
+    // islandora_multi_importer_form
+    case 'islandora_multi_importer_form':
+      islandora_multi_importer_form_restore_selections($form);
+      break;
+
+  }
+
+  return;
+}
+
+/**
+ * islandora_multi_importer_form_save_selections( )
+ *
+ * Addresses https://github.com/mnylc/islandora_multi_importer/issues/22
+ *
+ * @param $form_state - $form_state array.
+ * @param $form - The full $form array.
+ *
+ */
+function islandora_multi_importer_form_save_selections($form_state, $form) {
+  $keys = islandora_multi_importer_form_save_paths();
+
+  // If $form['step4'] exists...check $form_state['values'] looking for arrays with keys matching $keys
+  if (isset($form['step4'])) {
+    foreach ($keys as $k) {
+      $burst = explode('|', $k);
+      $count = count($burst);
+      $val = FALSE;
+
+      switch ($count) {
+        case 5:
+          list($a,$b,$c,$d,$e) = $burst;
+          if (isset($form_state['input']['cmodelmap'][$a][$b][$c][$d][$e])) {
+            $val = $form_state['input']['cmodelmap'][$a][$b][$c][$d][$e];
+          } else if (isset($form_state['values'][$a][$b][$c][$d][$e])) {
+            $val = $form_state['values'][$a][$b][$c][$d][$e];
+          }
+          break;
+        case 6:
+          list($a,$b,$c,$d,$e,$f) = $burst;
+          if (isset($form_state['input']['cmodelmap'][$a][$b][$c][$d][$e][$f])) {
+            $val = $form_state['input']['cmodelmap'][$a][$b][$c][$d][$e][$f];
+          } else if (isset($form_state['values'][$a][$b][$c][$d][$e][$f])) {
+            $val = $form_state['values'][$a][$b][$c][$d][$e][$f];
+          }
+          break;
+        case 7:
+          list($a,$b,$c,$d,$e,$f,$g) = $burst;
+          if (isset($form_state['input']['cmodelmap'][$a][$b][$c][$d][$e][$f][$g])) {
+            $val = $form_state['input']['cmodelmap'][$a][$b][$c][$d][$e][$f][$g];
+          } else if (isset($form_state['values'][$a][$b][$c][$d][$e][$f][$g])) {
+            $val = $form_state['values'][$a][$b][$c][$d][$e][$f][$g];
+          }
+          break;
+      }
+      if ($val) {
+        // Build a variable to save $val as the #default_value for $key
+        $name = "islandora_multi_key-$k";
+        variable_set($name, $val);
+      }
+    }
+  }
+  return;
+}
+
+/**
+ * islandora_multi_importer_form_restore_selections( )
+ *
+ * Addresses https://github.com/mnylc/islandora_multi_importer/issues/22
+ *
+ * @param $form - The full $form array.
+ *
+ */
+function islandora_multi_importer_form_restore_selections(&$form) {
```

Comment:
I need to test this. Looks pretty cool. Are you ok with me moving this actually to a class where we can save the settings (named) in DB?